reinforcement_learning/bindings/cs/rl.net/LiveModel.cs

```diff
@@ -51,8 +73,39 @@ public bool TryChooseRank(string actionId, string contextJson, RankingResponse r
 
         public bool TryReportOutcome(string actionId, float outcome, ApiStatus apiStatus = null)
         {
-            int result = LiveModelReportOutcome(this.NativeHandle, actionId, outcome, apiStatus.ToNativeHandleOrNullptr());
+            int result = LiveModelReportOutcomeF(this.NativeHandle, actionId, outcome, apiStatus.ToNativeHandleOrNullptr());
             return result == NativeMethods.SuccessStatus;
         }
+
+        public bool TryReportOutcome(string actionId, string outcomeJson, ApiStatus apiStatus = null)
+        {
+            int result = LiveModelReportOutcomeJson(this.NativeHandle, actionId, outcomeJson, apiStatus.ToNativeHandleOrNullptr());
+            return result == NativeMethods.SuccessStatus;
+        }
+
+        private event EventHandler<ApiStatus> BackgroundErrorInternal;
+
+        // TODO: This class need a pass to ensure thread-safety (or explicit declaration of non-thread-safe)
+        public event EventHandler<ApiStatus> BackgroundError
```

Comment:
It looks like there can only ever be one callback added? The convention is to be able to add any number, so this may be unexpected behavior for a consumer.

Reply:
The aggregation logic is invisible, because it is auto-implemented in the `BackgroundErrorInternal` event. If you note, we pass it the delegate (`value`) from `BackgroundError`. Then in the actual managed-side callback, `WrapStatusAndRaiseBackgroundError`, raises the -`Internal` event, rather than the outer one consumers of the API use to register.